### DISCUSSION TITLE
How to reset chezmoi

### DISCUSSION THREAD

**Discussion by Archie-2021** [twpayne/chezmoi#3247] [Q&A] [answered]
Hey guys, I am new to Chezmoi
I am trying to add my . dotfiles on a machine that are all symlinked to a centeral folder. It turns out adding the to chezmoi is tricky for directories. I think I messed up my repo when i added something via add command but removed it manually from chezmoi directory. 
Should I use remove the repo and restart fresh? Is there a command for it?

**Comment by twpayne** [comment, marked as answer]
If you were committing files to git then you can reset your dotfiles repo to a previous commit. See the git documentation for how to do this.

Otherwise, you can restore from backup.

Otherwise, if you have no git history and no backup, then you can reset the chezmoi source directory by removing it, i.e. `rm ~/.local/share/chezmoi`.

**Reply by Archie-2021** [reply]
Interesting, no I have not committed anything yet. I thought since I had added stuff via `chezmoi add` command there is a record for this mapping between the file in home directory and source directory and removing it manually would cause trouble later. If I can change stuff with just using git directly that is fine for me.

Another question , altho I know it maybe needs another thread. I want to use my dotfiles on multiple Machines and I want them to be different because I use those machines (with different OS) for different reasons? Should I use different branches for that or it is just templates? Tbh honest I have reading User-guide today and couldn't figure out anything substantial about templates and how they work.

**Reply by twpayne** [reply]
The [user guide](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/) answers your question directly. There's a complete [section on templating](https://www.chezmoi.io/user-guide/templating/).

**Reply by Archie-2021** [reply]
> The [user guide](https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/) answers your question directly. There's a complete [section on templating](https://www.chezmoi.io/user-guide/templating/).

Ok after reading it again it makes more sense now. Still kinda complicated language. Apparently syntax is Go , which is good for learning but I didn't know much about it.
